source/tomopy/prep/stripe.py

```diff
@@ -939,3 +941,151 @@ def _remove_stripe_based_interpolation(tomo, snr, size, drop_ratio, norm):
+    full and partial stripes of constant ot varying intensity. 
+    
+    .. versionadded:: 1.14
+
+    Parameters
+    ----------
+    inputData : ndarray
+        3D tomographic data of float32 data type, normalized [0,1] and given in
+        [angles, detY(depth), detX (horizontal)] axis orientation.
+    size : int, optional
+        The pixel size of the vertical 1D median filter to minimise false detections. Increase it if you have longer or full stripes in the data. 
+    radius : int, optional
+        The pixel size of the stencil to calculate the mean ratio between vertical and horizontal orientations. The larger values will enlarge the mask width.
+    ncore : int, optional
+        Number of cores that will be assigned to jobs. All cores will be used
+        if unspecified.
+
+    Returns
+    -------
+    ndarray
+        Weights for stripe's edges as a 3D array of float32 data type. 
+        The weights can be thresholded or passed to stripes_mask3d function to obtain a binary mask.
+
+    Raises
+    ------
+    ValueError
+        If the input array is not three dimensional.        
+    """ 
+    if ncore is None:
+        ncore = mproc.mp.cpu_count()
+
+    input_type = tomo.dtype
+    if (input_type != 'float32'):
+        tomo = dtype.as_float32(tomo)  # silent convertion to float32 data type
+    out = np.empty_like(tomo, order='C')
+    
+    if tomo.ndim == 3:
+        dz, dy, dx = tomo.shape
+        if (dz == 0) or (dy == 0) or (dx == 0):
+            raise ValueError("The length of one of dimensions is equal to zero")
+    else:
+        raise ValueError("The input array must be a 3D array")
+
+    if size <= 0 or size > dz //  2:
+        raise ValueError("The size of the filter should be larger than zero and smaller than the half of the vertical dimension")
+
+    # perform stripes detection
+    extern.c_stripes_detect3d(np.ascontiguousarray(tomo), 
+                              out, 
+                              size,
+                              radius,
+                              ncore,
+                              dx, dy, dz)
+    return out
+
+
+def stripes_mask3d(weights, 
+                  threshold = 0.6,
+                  min_stripe_length = 20,
+                  min_stripe_depth  = 10,
+                  min_stripe_width = 5,
+                  sensitivity_perc = 85.0,
+                  ncore=None):
+    """
+    Takes the result of the stripes_detect3d module as an input and generates a 
+    binary 3D mask with ones where stripes present. The method tries to eliminate
+    non-stripe features in data by checking the weight consistency in three directions.
+
+    .. versionadded:: 1.14
+
+    Parameters
+    ----------
+    weights : ndarray
+        3D weights array, a result of stripes_detect3d module given in
+        [angles, detY(depth), detX] axis orientation.
+    threshold : float, optional
+        Threshold for the given weights, the smaller values correspond to the stripes
```

Comment:
Thanks, fair point. I've re-written the description to: _This parameter defines what weights will be considered as potential candidates for stripes. It is important to remove as many false outliers as possible by taking the highest acceptable value. The lower value might lead to ignore the actual stripes and the higher one can generate false alarms._

Hopes it is clearer now.

Reply:
This is a 'less than' threshold or 'greater than' threshold? These explanations are contradictory because they claim raising the threshold both increases the number of false alarms and removes outliars?

Are lower valued weights more likely or less likely to be stripes?

Reply:
I understand how a bit confusing the description might sound.

_Are lower valued weights more likely or less likely to be stripes?_ The lower weights are **more** likely to be stripes.

 Basically if we take the threshold level too low, e.g., lower than 0.5 normally, then we will get the most prominent stripes only and the weak stripes will be mostly ignored. It is tricky to find the exactly right threshold value that would lead to both weak and prominent stripes detected equally well. So I'm suggesting taking the value when the false alarms just start to appear, hence "highest acceptable"...

Reply:
I slightly re-wrote that bit.